errorlint/lint.go

```diff
@@ -322,10 +466,91 @@ func LintErrorTypeAssertions(fset *token.FileSet, info *TypesInfoExt) []analysis
 			continue
 		}
 
-		lints = append(lints, analysis.Diagnostic{
+		diagnostic := analysis.Diagnostic{
 			Message: "type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors",
 			Pos:     typeAssert.Pos(),
-		})
+		}
+
+		// Create suggested fix for type assertion
+		targetType := exprToString(typeAssert.Type)
+		errExpr := exprToString(typeAssert.X)
+
+		parent := info.NodeParent[typeAssert]
+
+		// For assignment statements like: targetErr, ok := err.(*SomeError)
+		if assign, ok := parent.(*ast.AssignStmt); ok && len(assign.Lhs) == 2 {
+			if id, ok := assign.Lhs[0].(*ast.Ident); ok {
+				varName := id.Name
+
+				// If this is part of an if statement initialization
+				ifParent, isIfInit := info.NodeParent[assign].(*ast.IfStmt)
+				if isIfInit && ifParent.Init == assign {
+					// Handle special case for if statements
+					// Replace: if targetErr, ok := err.(*SomeError); ok {
+					// With:    if targetErr := new(SomeError); errors.As(err, targetErr) {
+
+					// Get the condition of the if statement (usually just "ok")
+					replacement := fmt.Sprintf("%s := new(%s);\nerrors.As(%s, %s)",
+						varName, targetType[1:], // Remove the leading * from the type name
+						errExpr, varName)
```

Comment:
I'm a bit worried the code could be without pointer


```
if targetErr, ok := err.(SomeError); ok {
```

I wouldn't use `targetType[1:]` but strings.TrimPrefix(targetType, "*")` 

Please note, it might be OK, maybe it's already covered and tested, but I'm always afraid by code that arbitrarily remove a character